config/v3_1/types/storage_test.go

```diff
@@ -227,3 +234,94 @@ func TestStorageValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestStorageValidateWarnings(t *testing.T) {
+	tests := []struct {
+		in  Storage
+		at  path.ContextPath
+		out error
+	}{
+		// test a disk with saved partitions with the same 'device' as a filesystem returns ErrPartitionsOverwritten
```

Comment:
Let's drop "saved".  This isn't coreos-installer-style partition saving, this is usually creating new partitions.